src/components/Navigator/QuickNavigationModal.vue

```diff
@@ -9,180 +9,368 @@
 -->
 
 <template>
-  <div class="quick-navigation-modal">
-    <div class="container">
+  <GenericModal
+    ref="modal"
+    pinContent
+    :showClose="false"
+    :visible="showQuickNavigationModal"
+    @update:visible="$emit('closeQuickNavigationModal')"
+  >
+    <div
+      class="quick-navigation"
+      @keydown.down.exact.prevent="focusNext"
+      @keydown.up.exact.prevent="focusPrev"
+      @keydown.enter.exact="handleKeyEnter"
+    >
       <div
-        class="close-icon"
-        @click="closeQuickNavigationModal()"
+        class="quick-navigation__container"
       >
-        <p>
-          Close
-        </p>
-      </div>
-      <input
-        ref="input"
-        class="filter"
-        v-model="userInput"
-        type="text"
-      />
-      <div>
+        <FilterInput
+          v-model="userInput"
+          class="quick-navigation__filter"
+          placeholder="Search symbols"
+          focusInputWhenCreated
+          focusInputWhenEmpty
+          @input="focusedIndex = 0"
+        >
+          <template #icon>
+            <div
+              class="quick-navigation__magnifier-icon-container"
+              :class="{ 'blue': userInput.length }"
+            >
+              <MagnifierIcon />
+            </div>
+          </template>
+          <template #input-menu-items>
+            <div
+              v-if="!userInput.length"
+              class="quick-navigation__open-key-container"
+            >
+              <kbd>Cmd+Shift+o</kbd>
+            </div>
+          </template>
+        </FilterInput>
         <div
-          v-for="(symbol, idx) in filteredSymbols"
-          class="symbol-match"
-          @click="closeQuickNavigationModal()"
-          :key="idx"
+          class="quick-navigation__match-list"
+          :class="{ 'active' : debouncedInput.length }"
         >
-          <Reference :url="symbol.path" :id="idx">
-            <span>
-              -
-              {{ symbol.title.slice(0, symbol.start) }}
-            </span>
-            <QuickNavigationHighlighter
-              :text="symbol.substring"
-              :matcherText="debouncedInput"
-            />
-            <span>
-              {{ symbol.title.slice(symbol.start + symbol.matchLength) }}
-            </span>
-          </Reference>
+          <div
+            v-if="noResultsWereFound"
+            class="no-results"
+          >
+            <p>
+              No results found.
+            </p>
+          </div>
+          <div
+            v-for="(symbol, index) in filteredSymbols"
+            :class="{ 'selected' : index == focusedIndex }"
+            :key="symbol.uid"
+            tabindex="0"
```

Comment:
Why this complication of adding tabindex to a div and not use the Reference's natural focusability, since its a link?

Reply:
I'm probably missing something here, but having the div tag wrapping `Reference` makes it necessary to use `tabindex="0"`. I tried to remove this external div and loop over `Reference` directly, but then the `@@focus.capture="focusIndex(index)"` logic doesn't get triggered.

Reply:
Because Reference is a Vue component, try with `@focus.native`

Reply:
Thanks for pointing this out! Resolved here 4bd2fd13e893ecb2b3dee9617a6ba7e430882f48